File: ticket-service/src/main/java/com/helpcore/ticket_service/entidades/dto/VerTicketDTO.java
package com.helpcore.ticket_service.entidades.dto;

import com.helpcore.ticket_service.entidades.Ticket;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class VerTicketDTO {
    private Integer id;
    private String codigoTicket;
    private String titulo;
    private String descripcion;
    private Ticket.Estado estado;
    private Ticket.Prioridad prioridad;
    private String codigoAlumno;
    private LocalDateTime fechaCreacion;
    private LocalDateTime fechaAsignacion;
    private LocalDateTime fechaResolucion;
    private LocalDateTime fechaCierre;

    private SolicitanteDTO solicitante;
    private AgenteDTO agente;
    private CategoriaDTO categoria;

    @Data
    public static class SolicitanteDTO {
        private Integer id;
        private String nombre;
        private String apellido;
        private String correo;
        private String telefono;
        private String dni;
        private String codigoAlumno;
        private SedeDTO sede;  // Agregar este campo
    }

    @Data
    public static class SedeDTO {
        private Integer id;
        private String nombre;
    }

    @Data
    public static class AgenteDTO {
        private Integer id;
        private String nombreCompleto;
        private String correo;
    }

    @Data
    public static class CategoriaDTO {
        private Integer id;
        private String nombre;
    }
}

File: helpcore-fe/src/app/dto/ver-ticket-dto.ts
export interface VerTicketDto {
  id: number;
  codigoTicket: string;
  titulo: string;
  descripcion: string;
  estado: EstadoTicket;
  prioridad: PrioridadTicket;
  codigoAlumno: string;
  fechaCreacion: string;
  fechaAsignacion?: string;
  fechaResolucion?: string;
  fechaCierre?: string;
  solicitante: Solicitante;
  agente?: Agente;
  categoria: Categoria;
  sede: Sede;
}

export interface Solicitante {
  id: number;
  nombre: string;
  apellido: string;
  correo: string;
  telefono: string;
  sede: Sede;
  dni: string;
  codigoAlumno: string;
}

export interface Agente {
  id: number;
  nombreCompleto: string;
  email: string;
}

export interface Categoria {
  id: number;
  nombre: string;
}

export interface Sede {
  id: number;
  nombre: string;
}

export enum EstadoTicket {
  NUEVO = 'NUEVO',
  EN_ATENCION = 'EN_ATENCION',
  RESUELTO = 'RESUELTO',
  CERRADO = 'CERRADO'
}

export enum PrioridadTicket {
  BAJA = 'BAJA',
  MEDIA = 'MEDIA',
  ALTA = 'ALTA',
  URGENTE = 'URGENTE'
}